cmd/image-builder/version.go

```diff
@@ -0,0 +1,36 @@
+package main
+
+import (
+	"fmt"
+	"runtime/debug"
+)
+
+// Get the version for this project based on the passed in string, the passed in string
+// can potentially be set to something externally (for example through build arguments).
+// When the string has any value other than the sentinel value "DEVEL" we pass it along
+// as-is. Otherwise we try and get information from the build info.
+func GetVersion(v string) string {
+	// buildversion was set at build time, don't do any special handling but just
+	// return it as-is
+	if v != "DEVEL" {
+		return v
+	} else {
+		// otherwise try to get it from the build information
+		return versionFromBuildInfo()
+	}
+}
+
+// Try and get a version (commit) from the build info included in the binary since Go 1.18.
+func versionFromBuildInfo() string {
+	version := "DEVEL"
```

Comment:
We have a very similar function in bib (c.f. https://github.com/osbuild/bootc-image-builder/blob/main/bib/cmd/bootc-image-builder/main.go#L587), having them similar seems nice, I personally think the "buildTime" and "buildTained" are a bit superficial but `build_revision: <rev>` seems nice and keeping it valid yaml is also nice (and we get it almost for free).

(and the test is in https://github.com/osbuild/bootc-image-builder/blob/main/test/test_opts.py#L137)

Reply:
FTR, we also have https://github.com/osbuild/osbuild-composer/blob/cedc351bbd5422d04e4df53551fc698d54346ca6/internal/common/constants.go#L30-L38